### PR TITLE
Feature/worker deployment

### PR DESCRIPTION
**Pull request by raonismaneoto**

### CODE REVIEW

File: install-worker.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+HOSTS_CONF_FILE="$(pwd)/conf-files/hosts.conf"
+ANSIBLE_HOSTS_FILE="$(pwd)/ansible/worker/hosts"
+ANSIBLE_CFG_FILE="$(pwd)/ansible/worker/ansible.cfg"
+
+# setup hosts
+WORKER_HOST_IP=$(grep ^worker_host $HOSTS_CONF_FILE | awk -F "=" '{print $2}')
+PATTERN_HELPER="\[worker-machine\]"
+WORKER_HOST_IP_PATTERN=$(grep -A1 $PATTERN_HELPER $ANSIBLE_HOSTS_FILE | tail -n 1)
+echo $WORKER_HOST_IP_PATTERN
+sed -i "s/$WORKER_HOST_IP_PATTERN/$WORKER_HOST_IP/" $ANSIBLE_HOSTS_FILE
```
Comment (wesleymonte) [inline install-worker.sh]: You can use `sed -i '/\[worker-machine\]/,/\[worker-machine:vars\]/{//!d}' ${ANSIBLE_HOSTS_FILE}` to delete everything between `[worker-machine]` and `[worker-machine:vars]`. That way you also won't need `# WORKER-PATTERN-HELPER #`

File: install-worker.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+WORKER_NODE_HOST_IP=$(grep ^worker_node_host $HOSTS_CONF_FILE | awk -F "=" '{print $2}')
+PATTERN_HELPER="\[worker-node-machine\]"
+WORKER_NODE_HOST_IP_PATTERN=$(grep -A1 $PATTERN_HELPER $ANSIBLE_HOSTS_FILE | tail -n 1)
+sed -i "s/$WORKER_NODE_HOST_IP_PATTERN/$WORKER_NODE_HOST_IP/" $ANSIBLE_HOSTS_FILE
```
Comment (wesleymonte) [inline install-worker.sh]: You can use `sed -i '/\[worker-node-machine\]/,/\worker-node-machine:vars\]/{//!d}' ${ANSIBLE_HOSTS_FILE}` to delete everything between [worker-node-machine] and [worker-node-machine:vars].

File: services/worker/deploy-script.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+check_docker_installation
+
+CONF_FILE_PATH="./worker-conf.json"
+TAG=${1-latest}
+IMAGE=raonismaneoto/arrebol-worker:$TAG
+ID=`jq .Id $CONF_FILE_PATH`
+ID=${ID%\"}
+ID=${ID#\"}
```
Comment (wesleymonte) [inline services/worker/deploy-script.sh]: Could you add comments explaining these lines? Or even separate into a function with a self-explanatory name.

File: services/worker/deploy-script.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    sudo systemctl start docker
+    sudo systemctl enable docker
+  fi
+}
+
+check_docker_installation
+
+CONF_FILE_PATH="./worker-conf.json"
+TAG=${1-latest}
+IMAGE=raonismaneoto/arrebol-worker:$TAG
```
Comment (wesleymonte) [inline services/worker/deploy-script.sh]: Remove the $TAG from IMAGE and add `readonly` to it.

File: services/worker/deploy-script.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  if [[ $EXIT_CODE -ne 0 ]]; then
+    sudo apt-get install docker.io
+    sudo systemctl start docker
+    sudo systemctl enable docker
+  fi
+}
+
+check_docker_installation
+
+CONF_FILE_PATH="./worker-conf.json"
```
Comment (wesleymonte) [inline services/worker/deploy-script.sh]: Add readonly.

File: services/worker/deploy-script.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+sudo docker pull $IMAGE
+sudo docker stop $CONTAINER_NAME
+sudo docker rm $CONTAINER_NAME
+sudo docker run --name $CONTAINER_NAME -tdi \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        -v /usr/bin/docker:/usr/bin/docker \
+        $IMAGE
+
+PROJECT_PATH="/go/src/github.com/ufcg-lsd/arrebol-pb-worker"
```
Comment (wesleymonte) [inline services/worker/deploy-script.sh]: Add readonly.

File: install-worker.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+ANSIBLE_SSH_PRIVATE_KEY_FILE=$(grep ^private_key_path $HOSTS_CONF_FILE | awk -F "=" '{print $2}')
+PRIVATE_KEY_FILE_PATH_PATTERN="ansible_ssh_private_key_file"
+sed -i "s#.*$PRIVATE_KEY_FILE_PATH_PATTERN=.*#$PRIVATE_KEY_FILE_PATH_PATTERN=$ANSIBLE_SSH_PRIVATE_KEY_FILE#g" $ANSIBLE_HOSTS_FILE
+
+#setup ansible.cfg
+REMOTE_USER=$(grep ^remote_host_user $HOSTS_CONF_FILE | awk -F "=" '{print $2}')
+PATTERN_HELPER="remote_user"
```
Comment (wesleymonte) [inline install-worker.sh:32]: I think something like `REMOTE_USER_PATTERN` is a better name for this variable.

**Review by wesleymonte — CHANGES_REQUESTED** [review]

File: install-worker.sh (hunk 1 above)
Comment (raonismaneoto) [reply install-worker.sh]: But how will I include the worker ip then?

File: install-worker.sh (hunk 7 above)
Comment (raonismaneoto) [reply install-worker.sh:32]: I think it is very expressive already, once each "block" has a pattern_help that changes according to its semantic. What do you think?

File: install-worker.sh (hunk 1 above)
Comment (wesleymonte) [reply install-worker.sh]: `sed -i "/\[worker-machine:vars\]/i ${HOST_IP}" ${ANSIBLE_HOSTS_FILE}`
This command will write the host address above [worker-machine: vars], which means adding at the end of [worker-machine].

File: install-worker.sh [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+sed -i "s#.*$PATTERN_HELPER = .*#$PATTERN_HELPER = $REMOTE_USER#g" $ANSIBLE_CFG_FILE
+
+#cd ansible/worker && ansible-playbook -vvv deploy.yml
```
Comment (wesleymonte) [inline install-worker.sh]: I think you accidentally commented on that line.

File: ansible/worker/hosts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+[worker-machine]
+#WORKER-PATTERN-HELPER#
+[worker-machine:vars]
+ansible_ssh_private_key_file=
+ansible_python_interpreter=/usr/bin/python3
+
+[worker-node-machine]
+#WORKER-NODE-PATTERN-HELPER#
+[worker-node-machine:vars]
```
Comment (wesleymonte) [inline ansible/worker/hosts]: The `#WORKER-PATTERN-HELPER#` tag is no longer needed. You can replace it with a blank line.

**Review by wesleymonte — CHANGES_REQUESTED** [review]

**Review by wesleymonte — APPROVED** [review]

**Pull request merged**